core/src/test/java/com/opensymphony/xwork2/ognl/OgnlValueStackTest.java

```diff
@@ -980,6 +1001,272 @@ public void testWarnAboutInvalidProperties() {
         assertEquals(null, stack.findValue("address.country.name", String.class));
     }
 
+   /**
+     * Test a default OgnlValueStackFactory and OgnlValueStack generated by it
+     * when a default configuration is used.
+     */
+    public void testOgnlValueStackFromOgnlValueStackFactoryDefaultConfig() {
+        OgnlValueStackFactory ognlValueStackFactory = (OgnlValueStackFactory)
+                container.getInstance(ValueStackFactory.class, Container.DEFAULT_NAME);
+
+        OgnlValueStack ognlValueStack = (OgnlValueStack) ognlValueStackFactory.createValueStack();
```

Comment:
I see there is already `OgnlValueStackTest.createValueStack` available method which is used in other tests. I think it's a good practice to keep current existing codes conduct and use it here instead.

P.S. It seems `OgnlValueStackTest.createValueStack` needs a fix about not used `allowStaticFieldAccess` parameter. Please feel free also if any improvement to it was needed to support your newly added tests :) thanks in advance!

Reply:
Hello @yasserzamani .

I fixed the issue with `createValueStack()` not setting the `allowStaticFieldAccess` parameter.

However since the other tests were intentionally using an (_intentionally_) different creation strategy using `OgnlValueStackFactory` directly to generate the `OgnlValueStack` instance for testing I couldn't see a good way to modify the existing `createValueStack()` method.

Instead I refactored some of the repetitive code into two separate generator methods for `OgnlValueStackFactory` instances:
- `createDefaultValueStackFactory()`, which generates based on the container configuration (from `setUp()`)
- `createRawValueStackFactory()`, which generates from raw factories (independent of the configuration from `setUp()` to better isolate the state testing of the flags) with boolean control flags to set the state (including the "clear-static-flags-from-properties").

It makes the overall code cleaner and promotes re-use (one of which was to add one additional test with flags set to match the default-from-container-state, to verify both behave the same).

Hopefully the changes look better and make sense ?